extensions/schema-compare/src/test/utils.test.ts

```diff
@@ -48,8 +58,58 @@ describe('utils: Tests to verify verifyConnectionAndGetOwnerUri', function (): v
 		testDatabaseEndpoint.connectionDetails = undefined;
 
-		ownerUri = await verifyConnectionAndGetOwnerUri(testDatabaseEndpoint, 'test');
+		ownerUri = await verifyConnectionAndGetOwnerUri(testDatabaseEndpoint, 'test', testContext.apiWrapper.object);
 
 		should(ownerUri).equal(undefined);
 	});
 });
+
+describe('utils: In-depth tests to verify verifyConnectionAndGetOwnerUri', function (): void {
+	before(async function (): Promise<void> {
+		testContext = createContext();
+	});
+
+	it('Should throw an error asking to make a connection', async function (): Promise<void> {
+		let getConnectionsResults: azdata.connection.ConnectionProfile[] = [];
+		let connection  = {...mockConnectionResult};
+		let testDatabaseEndpoint: mssql.SchemaCompareEndpointInfo = {...mockDatabaseEndpoint};
+		testDatabaseEndpoint.connectionDetails = {...mockConnectionInfo};
+		const getConnectionString = loc.getConnectionString('test');
+
+		testContext.apiWrapper.setup(x => x.connect(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => { return Promise.resolve(connection); });
+		testContext.apiWrapper.setup(x => x.getUriForConnection(TypeMoq.It.isAny())).returns(() => { return Promise.resolve(undefined); });
+		testContext.apiWrapper.setup(x => x.getConnections(TypeMoq.It.isAny())).returns(() => { return Promise.resolve(getConnectionsResults); });
+		testContext.apiWrapper.setup(x => x.showWarningMessage(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+
+		await shouldThrowSpecificError(async () => await verifyConnectionAndGetOwnerUri(testDatabaseEndpoint, 'test', testContext.apiWrapper.object), getConnectionString);
+	});
+
+	it('Should throw an error for login failure', async function (): Promise<void> {
+		let getConnectionsResults: azdata.connection.ConnectionProfile[] = [{...mockConnectionProfile}];
+		let connection  = {...mockConnectionResult};
+		let testDatabaseEndpoint: mssql.SchemaCompareEndpointInfo = {...mockDatabaseEndpoint};
+		testDatabaseEndpoint.connectionDetails = {...mockConnectionInfo};
+
+		testContext.apiWrapper.setup(x => x.connect(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => { return Promise.resolve(connection); });
+		testContext.apiWrapper.setup(x => x.getUriForConnection(TypeMoq.It.isAny())).returns(() => { return Promise.resolve(undefined); });
+		testContext.apiWrapper.setup(x => x.getConnections(TypeMoq.It.isAny())).returns(() => { return Promise.resolve(getConnectionsResults); });
+		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+
+		await shouldThrowSpecificError(async () => await verifyConnectionAndGetOwnerUri(testDatabaseEndpoint, 'test', testContext.apiWrapper.object), connection.errorMessage);
+	});
+
+	it('Should not throw an error for login failure with openConnectionDialog', async function (): Promise<void> {
```

Comment:
should or should not? line 113 seems like it does throw.

Reply:
My bad, it should be "should". I will remove the "not".